src/relax/analysis/well_formed.cc

```diff
@@ -56,6 +56,14 @@
  *           * The op or args fields of Call nodes
  *           * Inside the fields of Tuple nodes
  *    13. Expr always has checked_type_ (with the exception of Op).
+ *    14. DataflowBlocks may not contain If nodes.
+ *    15. DataflowBlocks may not contain calls to impure functions or operators
+ *        (only checked if check_struct_info is true).
+ *    16. If a function is annotated as pure (kIsPure is true)
+ *        and purity is not forced (kForcePure is true),
+ *        the body may not contain any impure call
+ *        (only checked if check_struct_info is true).
+ *    17. If a function's purity is forced, kForcePure cannot be true
```

Comment:
I don't understand `17`. If the function's purity is forced, wouldn't kForcePure have to be set as `true`?

Reply:
Sorry, that was a typo. `kIsPure` cannot be false in that case. Will fix.